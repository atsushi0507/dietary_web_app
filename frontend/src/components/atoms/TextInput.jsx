import React from "react";
import styled from "styled-components";


const TextInput = ({
    placeholder = "",  // イコールの前後にスペース
    name = "",
    type = "text",
    required = false,
    value = "",
    onChange = () => {},
    pattern = ""
}) => {
    return (
        <SInput
            placeholder={placeholder}
            name={name}
            type={type}
            required={required}
            value={value}
            onChange={onChange}
            pattern={pattern}
        />
    );
};

export default TextInput;

const SInput = styled.input`

`