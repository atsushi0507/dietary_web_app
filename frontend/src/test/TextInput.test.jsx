import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import TextInput from '../components/atoms/TextInput';
// import '@testing-library/jest-dom/extend-expect';

describe('TextInput component', () => {

    test('renders with the correct placeholder', () => {
        render(<TextInput placeholder="Enter your name" />);
        const inputElement = screen.getByPlaceholderText('Enter your name');
        expect(inputElement).toBeInTheDocument();
    });

    test('sets the correct name attribute', () => {
        render(<TextInput name="username" />);
        const inputElement = screen.getByRole('textbox');
        expect(inputElement).toHaveAttribute('name', 'username');
    });

    test('sets the correct type attribute', () => {
        render(<TextInput type="password" placeholder="Password" />);
        const inputElement = screen.getByPlaceholderText('Password');
        expect(inputElement).toHaveAttribute('type', 'password');
    });


    test('marks the input as required', () => {
        render(<TextInput required={true} />);
        const inputElement = screen.getByRole('textbox');
        expect(inputElement).toBeRequired();
    });

    test('displays the correct value', () => {
        render(<TextInput value="Test Value" onChange={() => {}} />);
        const inputElement = screen.getByDisplayValue('Test Value');
        expect(inputElement).toBeInTheDocument();
    });



    test('sets the correct pattern attribute', () => {
        render(<TextInput pattern="[A-Za-z]{3,}" />);
        const inputElement = screen.getByRole('textbox');
        expect(inputElement).toHaveAttribute('pattern', '[A-Za-z]{3,}');
    });

    test('renders with default props', () => {
        render(<TextInput />);
        const inputElement = screen.getByRole('textbox');
        expect(inputElement).toHaveAttribute('type', 'text'); // デフォルトの type
        expect(inputElement).toHaveAttribute('placeholder', ''); // デフォルトの placeholder
    });


    test('renders with an empty value', () => {
        render(<TextInput value="" />);
        const inputElement = screen.getByRole('textbox');
        expect(inputElement).toHaveValue(''); // 空の値が正しく表示されているか確認
    });

    test('matches the snapshot', () => {
        const { asFragment } = render(<TextInput />);
        expect(asFragment()).toMatchSnapshot(); // スナップショットテスト
    });
});


