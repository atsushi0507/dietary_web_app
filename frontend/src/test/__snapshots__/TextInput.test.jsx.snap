// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component matches the snapshot 1`] = `
<DocumentFragment>
  <input
    class="sc-blHHSb dUWAsA"
    name=""
    pattern=""
    placeholder=""
    type="text"
    value=""
  />
</DocumentFragment>
`;
